runtime/macros/src/entry.rs

```diff
@@ -43,7 +43,7 @@ pub fn main(_args: TokenStream, input: TokenStream) -> Result<TokenStream, Error
     }
 
     Ok(quote! {
-        const _: fn() = main;
+        const _: fn() = ::std::main;
```

Comment:
This constant is supposed to check the type of the main function, e.g. against `extern` and presence of return value.

Therefore I don't follow this change